src/components/generic/generic.cpp

```diff
@@ -28,6 +36,10 @@ Subtype Generic::subtype() {
     return Subtype(RDK, COMPONENT, "generic");
 }
 
+ProtoType Generic::do_command(std::unordered_map<std::string, ProtoType*> command) {
```

Comment:
I don't _think_ that should be because of `Generic` having an undefined virtual function (that's a pretty normal pattern), but there may be some stuff going on here with circular dependencies or somesuch...

Reply:
Did some more investigation: we can just add `= 0` to mark methods as "purely" virtual and not require a default implementation. See this SO [post](https://stackoverflow.com/questions/8931612/do-all-virtual-functions-need-to-be-implemented-in-derived-classes).

It seems like you can’t just write `virtual foo()` in an abstract class and not define it _anywhere_. A “pure” virtual method is something like `virtual foo() = 0` (must be overridden by derived classes). A “non-pure” virtual method is something like `virtual foo() {}` (derived classes can rely on the default implementation). I think we want the former for `Generic::do_command`, `ResourceSubtype::create_resource_server` and `ResourceSubtype::create_rpc_client`.